infrastructure/testdriver/set_timezone.html

```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>TestDriver set_time_zone method</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+
+<script>
+function defaultTimeZone() {
+  return (new Intl.DateTimeFormat()).resolvedOptions().timeZone;
+}
+async_test(t => {
+  let timeZone = "Asia/Taipei";
+  test_driver
+    .set_time_zone(timeZone)
+    .then(() => {
+      assert_equals(defaultTimeZone(), timeZone);
+      t.done()
+    })
+    .catch(() => assert_unreached("set_time_zone failed"));
+});
```

Comment:
This test passes in Taipei even if `set_time_zone` is a no-op. Is this a problem? If we wanted to avoid this pitfall, we could detect the case where `defaultTimeZone()` returns 'Asia/Taipei'` before running the test, and use a different timezone in that case. Same for the other tests. Or, we could do what you do in another test in this patch: set a specific timezone before running the test.